subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junit/JUnitCategoriesCoverageIntegrationSpec.groovy

```diff
@@ -79,7 +93,26 @@ class JUnitCategoriesCoverageIntegrationSpec extends JUnitMultiVersionIntegratio
         result.testClass("org.gradle.SomeLocaleTests").assertTestsExecuted('ok1 [de]', 'ok1 [en]', 'ok1 [fr]')
     }
 
+    def canCombineTagsWithCustomExtension() {
+        Assume.assumeTrue(isJupiter())
+
+        when:
+        run('test')
+
+        then:
+        DefaultTestExecutionResult result = new DefaultTestExecutionResult(testDirectory)
+        result.assertTestClassesExecuted('org.gradle.SomeLocaleTests')
+        result.testClass("org.gradle.SomeLocaleTests").assertTestCount(3, 0, 0)
+        result.testClass("org.gradle.SomeLocaleTests").assertTestsExecuted(
+            ['ok1(Locale)[1]', 'French'] as String[],
```

Comment:
Probably not - it was more that the existing method wouldn't work even if we only looked for the name (because of the way it does the assertion).  Another approach could be to just change how the assertion works and only check the test name.